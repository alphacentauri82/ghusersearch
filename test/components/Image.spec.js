import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {
  StyleSheetTestUtils,
} from 'aphrodite';

import Image from 'components/Image';

describe('Component: Image', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <Image src="test.png" />
    );
    expect(component).toMatchSnapshot();
  });

  it('should hide the spinner if the image is loaded', () => {
    const wrapper = shallow(
      <Image src="test.png" />
    );
    wrapper.setState({status: 'loaded'});
    expect(wrapper.html()).toMatchSnapshot();
  });

});
