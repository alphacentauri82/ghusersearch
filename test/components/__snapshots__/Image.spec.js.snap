// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Image should hide the spinner if the image is loaded 1`] = `"<div class=\\"Image_e296pg\\" style=\\"min-height:60px\\"><img alt=\\"\\" class=\\"u-imgResponsive\\" src=\\"test.png\\"/></div>"`;

exports[`Component: Image should render 1`] = `
<div
  className="Image_e296pg"
  style={
    Object {
      "minHeight": 60,
    }
  }
>
  <img
    alt=""
    className="u-imgResponsive"
    onLoad={[Function]}
    src="test.png"
  />
  <svg />
</div>
`;
